making_change/making_change.py

```diff
@@ -2,16 +2,60 @@
 
 import sys
 
+# try with largest denomination
+# reduce number of largest denomination being used
+# for each reduction, fill up the rest with second denomination
+# if only pennies left, return 1
+
+
+def make_change(amount, denominations, cache):
+    if amount < 0:
+        return 0
+    elif amount == 0:
```

Comment:
You can combine the two `elif` statements as `elif amount == 0 or len(denominations) == 1`